contracts/RestrictedNORI.sol

```diff
@@ -167,6 +168,17 @@ contract RestrictedNORI is
    */
   bytes32 public constant TOKEN_REVOKER_ROLE = keccak256("TOKEN_REVOKER_ROLE");
 
+  /**
+   * @notice Accounting for the maximum amount of RestrictedNORI that should be mintable manually.
+   * @dev In the case of a non-ERC1155-compatible supplier wallet address, minting RestrictedNORI during a
+   * purchase will fail and cause an event to be emitted. This variable is used to track the maximum amount
+   * of RestrictedNORI that can be redmedially minted to a supplier's compatible address to avoid over-
+   * minting the wrapper token and failing to have enough RestrictedNORI backed by wrapped NORI.
+   * TODO This variable should be used to enforce the maximum number of tokens that can ever be minted manually,
+   * and should be decremented when tokens are minted manually, which is not yet implemented.
+   */
+  uint256 private _maxManualMintable;
```

Comment:
For the sake of the strongest possible safeguards against minting Restricted NORI, it might make sense to turn this into a mapping from `(address -> uint256)`, so that we know specifically who is owed how much. That way, when we mint Restricted NORI, we prevent ourselves from accidentally minting too much for any given supplier.

Right now, it becomes difficult to safeguard against that possibility in the event that more than one supplier is responsible for incrementing this counter.

Reply:
yeah good idea.... waffled on this a little bit in favor of simplicity but it's not that useful that way. went with the mapping

Reply:
note that there's not currently a way to sum across all of the addresses because we're not using an enumerable mapping, so we don't know which entries the mapping contains (though these addresses can be retrieved by events, so I don't think we need to further burden this contract in the case that someday we want to understand the *total* deficit of this contract... not only is this scenario exceedingly rare, but there are other avenues to getting the complete set of affected addresses by events and creating the sum using the getter multiple times)